src/test/java/io/pravega/connectors/flink/source/reader/FlinkPravegaSplitReaderTest.java

```diff
@@ -157,10 +128,42 @@ private PravegaSplitReader createSplitReader(int subtaskId, String readerGroupNa
                 subtaskId);
     }
 
+    private PravegaSplitReader createTestableSplitReader() throws Exception {
+        return new TestablePravegaSplitReader(
+                "scope",
+                null,
+                "rg",
+                1);
+    }
+
     private static void createReaderGroup(String readerGroupName, String streamName) throws Exception {
         ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SETUP_UTILS.getScope(), SETUP_UTILS.getClientConfig());
         Stream stream = Stream.of(SETUP_UTILS.getScope(), streamName);
         readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().stream(stream).disableAutomaticCheckpoints().build());
         readerGroupManager.close();
     }
+
+    /**
+     * A Pravega split reader subclass for test purposes.
+     */
+    private static class TestablePravegaSplitReader extends PravegaSplitReader {
```

Comment:
We can name it as a `BadSplitReader`(or maybe like invalid, dummy) to indicate this class is used for negative cases. `Testable` would make people think it is for positive cases.
One more suggestion, add some more details on what is the purpose.(to test unsuccessful resource cleanup)